packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1752,30 +1759,62 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
         assert(this._orderSequentiallyCalls === 0,
             0x24c /* "Cannot call `flush()` from `orderSequentially`'s callback" */);
 
-        if (!this.deltaSender) {
-            return;
-        }
-
-        // Let the PendingStateManager know that there was an attempt to flush messages.
-        // Note that this should happen before the `this.needsFlush` check below because in the scenario where we are
-        // not connected, `this.needsFlush` will be false but the PendingStateManager might have pending messages and
-        // hence needs to track this.
-        this.pendingStateManager.onFlush();
+        const batch = this.batchManager.popBatch();
 
         // If flush has already been called then exit early
-        if (!this.needsFlush) {
-            return;
+        const length = batch.length;
+
+        if (length > 1) {
+            batch[0].metadata = { ...batch[0].metadata, batch: true };
+            batch[length - 1].metadata = { ...batch[length - 1].metadata, batch: false };
```

Comment:
Should we remove setting batch to false in ScheduleManager's prepareSend handler if we're going to do it here?

Reply:
we can't for back-compat reasons - we will have to support old loader version for very long time!
we will have to live with that, including rolling out features only for new code paths.

Reply:
Ah, just realized that "prepareSend" is emitted by the loader so we'll need both for compat to get batch metadata properly set